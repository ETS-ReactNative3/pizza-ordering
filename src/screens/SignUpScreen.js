import React, {Component} from 'react';

import {
  View,
  Text,
  TouchableOpacity,
  ImageBackground,
  Dimensions,
} from 'react-native';
import {dimensionStyles} from '../resources/dimension.js';
import {textStyle} from '../resources/textStyle.js';
import {StringInput} from '../components/StringInput.js';
import postUserAPI from '../repository/postUserAPI.js';
import Icon from 'react-native-vector-icons/FontAwesome';
import background from '../../assets/background.jpg';

export default class SignUpScreen extends Component {
  state = {
    isLoading: false,
    phone: '',
    password: '',
    confirmPassword: '',
    name: '',
    address: '',
  };

  navigateBack = () => {
    const {navigation} = this.props;
    navigation.goBack();
  };

  CreateSignUpData = () => {
    const {phone, password, confirmPassword, name, address} = this.state;
    const phone_regex = /((09|03|07|08|05)+([0-9]{8,9})\b)/g;
    const password_regex = /^(?=.*\d)(?=.*[A-Za-z]).{6,20}$/;
    const checkPhone = phone_regex.test(phone);
    const checkPassword = password_regex.test(password);
    if (phone === '') {
      alert('Phone is required');
    } else if (checkPhone === false) {
      alert('Please enter a valid phone number');
    } else if (password === '') {
      alert('Password is required');
    } else if (password.length < 6 || password.length > 20) {
      alert('Password length must be between 6 and 20');
    } else if (checkPassword === false) {
      alert('Password must contain number and character');
    } else if (password !== confirmPassword) {
      alert('Confirm password and password must be the same');
    } else if (name === '') {
      alert('Name is required');
    } else if (address === '') {
      alert('Addess is required');
    } else {
      let SignUpData = {
        phone: phone,
        password: password,
        name: name,
        address: address,
      };
      this.setState({isLoading: true});
      this.SignUp(SignUpData);
    }
  };

  SignUp = async SignUpData => {
    try {
      let response = await postUserAPI('/user', SignUpData);
      //console.log(response.data)
      alert('Sign up successfully!');
      this.navigateBack();
    } catch (errorMessage) {
      alert(errorMessage);
      console.log(errorMessage);
    }
    this.setState({isLoading: false});
  };

  render() {
    const {navigation} = this.props;
    const {phone, password, confirmPassword, name, address} = this.state;

    return (
      <View style={dimensionStyles.container}>
        <ImageBackground
          style={dimensionStyles.HomeImg}
          source={background}
          resizeMode="cover"
        />
        <TouchableOpacity
          style={dimensionStyles.goBackIconSignUpScreen}
          onPress={this.navigateBack}>
          <Icon name="angle-left" size={30} color="#FFFFFF" />
        </TouchableOpacity>
        <View style={dimensionStyles.containerCenter}>
          <Text
            style={{
              fontSize: 40,
              color: '#FFFFFF',
              marginBottom: 0.05 * Dimensions.get('window').height,
            }}>
            Join us!
          </Text>
          <StringInput
            text={phone}
            onTextChange={text => this.setState({phone: text})}
            placeholder={'Your phone number'}
            keyboard={'number-pad'}
            isSecure={false}
          />
          <StringInput
            text={password}
            onTextChange={text => this.setState({password: text})}
            placeholder={'Password'}
            keyboard={'default'}
            isSecure={true}
          />
          <StringInput
            text={confirmPassword}
            onTextChange={text => this.setState({confirmPassword: text})}
            placeholder={'Confirm password'}
            keyboard={'default'}
            isSecure={true}
          />
          <StringInput
            text={name}
            onTextChange={text => this.setState({name: text})}
            placeholder={'Your name'}
            keyboard={'default'}
            isSecure={false}
          />
          <StringInput
            text={address}
            onTextChange={text => this.setState({address: text})}
            placeholder={'Address'}
            keyboard={'default'}
            isSecure={false}
          />
          {this.state.isLoading ? (
            <TouchableOpacity
              style={dimensionStyles.LogInButton}
              disabled={true}>
              <Text style={textStyle.orderNowButton}>Please wait...</Text>
            </TouchableOpacity>
          ) : (
            <TouchableOpacity
              style={dimensionStyles.LogInButton}
              onPress={() => {
                this.CreateSignUpData();
              }}>
              <Text style={textStyle.orderNowButton}>Sign up</Text>
            </TouchableOpacity>
          )}
        </View>
      </View>
    );
  }
}
