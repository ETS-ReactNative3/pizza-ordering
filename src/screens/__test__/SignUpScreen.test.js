import React from 'react';
import {shallow, mount} from 'enzyme';
import configureStore from 'redux-mock-store';
import SignUpScreen from './../SignUpScreen.js';
import {Provider} from 'react-redux';

describe('sign up screen test', () => {
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });
  afterEach(() => {
    console.error = origConsole;
  });

  const initState = {};
  const mockStore = configureStore();
  let store, wrapper, props, instance;

  beforeEach(() => {
    props = {navigation: {goBack: jest.fn()}};
    store = mockStore(initState);
    wrapper = mount(
      <Provider store={store}>
        <SignUpScreen {...props} />
      </Provider>,
    );
    instance = wrapper.find('SignUpScreen').instance();
  });

  it('should render snapshot test for sign up screen', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should sign up with valid phone, password, name and address', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '123456789abc',
      confirmPassword: '123456789abc',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    const spy = jest.spyOn(instance, 'SignUp');
    instance.forceUpdate(); //call forceUpdate to attach the spy function to the instance

    signUpButton.at(0).simulate('click');
    expect(spy).toBeCalled();
  });

  it('should alert if phone is empty', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '',
      password: '123456789abc',
      confirmPassword: '123456789abc',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith('Phone is required');
  });

  it('should alert if phone is invalid', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '123456789',
      password: '123456789abc',
      confirmPassword: '123456789abc',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith('Please enter a valid phone number');
  });

  it('should alert if password is empty', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '',
      confirmPassword: '',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith('Password is required');
  });

  it('should alert if password length is not between 6 and 20', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '12a',
      confirmPassword: '',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith(
      'Password length must be between 6 and 20',
    );
  });

  it('should alert if password is not contain number and character', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '1234567',
      confirmPassword: '1234567',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith(
      'Password must contain number and character',
    );
  });

  it('should alert if confirm password is not same as password', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '1234567a',
      confirmPassword: '1234567ab',
      name: 'John Doe',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith(
      'Confirm password and password must be the same',
    );
  });

  it('should alert if name is empty', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '123456789abc',
      confirmPassword: '123456789abc',
      name: '',
      address: '247 Pasteur St.',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith('Name is required');
  });

  it('should alert if address is empty', () => {
    wrapper.find('SignUpScreen').setState({
      phone: '09755516009',
      password: '123456789abc',
      confirmPassword: '123456789abc',
      name: 'John Doe',
      address: '',
    });

    const signUpButton = wrapper.findWhere(node => {
      return node.text() === 'Sign up' && typeof node.type() === 'string';
    });

    window.alert = jest.fn();
    signUpButton.at(0).simulate('click');
    expect(window.alert).toBeCalledWith('Addess is required');
  });
});
