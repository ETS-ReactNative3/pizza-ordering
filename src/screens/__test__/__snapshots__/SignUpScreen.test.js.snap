// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign up screen test should render snapshot test for sign up screen 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SignUpScreen)
    navigation={
      Object {
        "goBack": [MockFunction],
      }
    }
  >
    <SignUpScreen
      navigation={
        Object {
          "goBack": [MockFunction],
        }
      }
      setUserToken={[Function]}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <ImageBackground
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/background.jpg",
              }
            }
            style={
              Object {
                "height": 1334,
                "position": "absolute",
                "width": 750,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 1334,
                  "position": "absolute",
                  "width": 750,
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "height": 1334,
                    "position": "absolute",
                    "width": 750,
                  }
                }
              >
                <Image
                  resizeMode="cover"
                  source={
                    Object {
                      "testUri": "../../../assets/background.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": 1334,
                        "width": 750,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    resizeMode="cover"
                    source={
                      Object {
                        "testUri": "../../../assets/background.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "height": 1334,
                          "width": 750,
                        },
                        undefined,
                      ]
                    }
                  />
                </Image>
              </View>
            </View>
          </ImageBackground>
          <ForwardRef
            onPress={[Function]}
            style={
              Object {
                "height": 30,
                "left": 10,
                "position": "absolute",
                "top": 10,
                "width": 30,
              }
            }
          >
            <TouchableOpacity
              hostRef={null}
              onPress={[Function]}
              style={
                Object {
                  "height": 30,
                  "left": 10,
                  "position": "absolute",
                  "top": 10,
                  "width": 30,
                }
              }
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "height": 30,
                      "left": 10,
                      "position": "absolute",
                      "top": 10,
                      "width": 30,
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "height": 30,
                        "left": 10,
                        "position": "absolute",
                        "top": 10,
                        "width": 30,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "height": 30,
                        "left": 10,
                        "opacity": 1,
                        "position": "absolute",
                        "top": 10,
                        "width": 30,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 30,
                          "left": 10,
                          "opacity": 1,
                          "position": "absolute",
                          "top": 10,
                          "width": 30,
                        }
                      }
                    >
                      <Icon
                        allowFontScaling={false}
                        color="#FFFFFF"
                        name="angle-left"
                        size={30}
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#FFFFFF",
                                "fontSize": 30,
                              },
                              undefined,
                              Object {
                                "fontFamily": "FontAwesome",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 30,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "FontAwesome",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            ï„„
                          </Text>
                        </Text>
                      </Icon>
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 40,
                    "marginBottom": 66.7,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 40,
                      "marginBottom": 66.7,
                    }
                  }
                >
                  Join us!
                </Text>
              </Text>
              <StringInput
                isSecure={false}
                keyboard="number-pad"
                onTextChange={[Function]}
                placeholder="Your phone number"
                text=""
              >
                <Component
                  allowFontScaling={true}
                  keyboardType="number-pad"
                  onChangeText={[Function]}
                  placeholder="Your phone number"
                  rejectResponderTermination={true}
                  secureTextEntry={false}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 15,
                      "height": 45,
                      "marginBottom": 3,
                      "marginTop": 3,
                      "width": 525,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    placeholder="Your phone number"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "fontSize": 15,
                        "height": 45,
                        "marginBottom": 3,
                        "marginTop": 3,
                        "width": 525,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </StringInput>
              <StringInput
                isSecure={true}
                keyboard="default"
                onTextChange={[Function]}
                placeholder="Password"
                text=""
              >
                <Component
                  allowFontScaling={true}
                  keyboardType="default"
                  onChangeText={[Function]}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 15,
                      "height": 45,
                      "marginBottom": 3,
                      "marginTop": 3,
                      "width": 525,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="default"
                    onChangeText={[Function]}
                    placeholder="Password"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "fontSize": 15,
                        "height": 45,
                        "marginBottom": 3,
                        "marginTop": 3,
                        "width": 525,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </StringInput>
              <StringInput
                isSecure={true}
                keyboard="default"
                onTextChange={[Function]}
                placeholder="Confirm password"
                text=""
              >
                <Component
                  allowFontScaling={true}
                  keyboardType="default"
                  onChangeText={[Function]}
                  placeholder="Confirm password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 15,
                      "height": 45,
                      "marginBottom": 3,
                      "marginTop": 3,
                      "width": 525,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="default"
                    onChangeText={[Function]}
                    placeholder="Confirm password"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "fontSize": 15,
                        "height": 45,
                        "marginBottom": 3,
                        "marginTop": 3,
                        "width": 525,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </StringInput>
              <StringInput
                isSecure={false}
                keyboard="default"
                onTextChange={[Function]}
                placeholder="Your name"
                text=""
              >
                <Component
                  allowFontScaling={true}
                  keyboardType="default"
                  onChangeText={[Function]}
                  placeholder="Your name"
                  rejectResponderTermination={true}
                  secureTextEntry={false}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 15,
                      "height": 45,
                      "marginBottom": 3,
                      "marginTop": 3,
                      "width": 525,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="default"
                    onChangeText={[Function]}
                    placeholder="Your name"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "fontSize": 15,
                        "height": 45,
                        "marginBottom": 3,
                        "marginTop": 3,
                        "width": 525,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </StringInput>
              <StringInput
                isSecure={false}
                keyboard="default"
                onTextChange={[Function]}
                placeholder="Address"
                text=""
              >
                <Component
                  allowFontScaling={true}
                  keyboardType="default"
                  onChangeText={[Function]}
                  placeholder="Address"
                  rejectResponderTermination={true}
                  secureTextEntry={false}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "fontSize": 15,
                      "height": 45,
                      "marginBottom": 3,
                      "marginTop": 3,
                      "width": 525,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="default"
                    onChangeText={[Function]}
                    placeholder="Address"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "fontSize": 15,
                        "height": 45,
                        "marginBottom": 3,
                        "marginTop": 3,
                        "width": 525,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </Component>
              </StringInput>
              <ForwardRef
                onPress={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#0b2031",
                    "borderRadius": 5,
                    "height": 45,
                    "justifyContent": "center",
                    "marginTop": 3,
                    "width": 525,
                  }
                }
              >
                <TouchableOpacity
                  hostRef={null}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#0b2031",
                      "borderRadius": 5,
                      "height": 45,
                      "justifyContent": "center",
                      "marginTop": 3,
                      "width": 525,
                    }
                  }
                >
                  <ForwardRef(AnimatedComponentWrapper)
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "backgroundColor": "#0b2031",
                          "borderRadius": 5,
                          "height": 45,
                          "justifyContent": "center",
                          "marginTop": 3,
                          "width": 525,
                        },
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                  >
                    <AnimatedComponent
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "backgroundColor": "#0b2031",
                            "borderRadius": 5,
                            "height": 45,
                            "justifyContent": "center",
                            "marginTop": 3,
                            "width": 525,
                          },
                          Object {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignSelf": "center",
                            "backgroundColor": "#0b2031",
                            "borderRadius": 5,
                            "height": 45,
                            "justifyContent": "center",
                            "marginTop": 3,
                            "opacity": 1,
                            "width": 525,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignSelf": "center",
                              "backgroundColor": "#0b2031",
                              "borderRadius": 5,
                              "height": 45,
                              "justifyContent": "center",
                              "marginTop": 3,
                              "opacity": 1,
                              "width": 525,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#FFFFFF",
                                "fontSize": 20,
                                "letterSpacing": 0.16,
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#FFFFFF",
                                  "fontSize": 20,
                                  "letterSpacing": 0.16,
                                  "textAlign": "center",
                                }
                              }
                            >
                              Sign up
                            </Text>
                          </Text>
                          <PressabilityDebugView
                            color="cyan"
                          />
                        </View>
                      </View>
                    </AnimatedComponent>
                  </ForwardRef(AnimatedComponentWrapper)>
                </TouchableOpacity>
              </ForwardRef>
            </View>
          </View>
        </View>
      </View>
    </SignUpScreen>
  </Connect(SignUpScreen)>
</Provider>
`;
